src/Common/tests/System/Net/VirtualNetwork/VirtualNetwork.cs

```diff
@@ -36,7 +36,24 @@ public void ReadFrame(bool server, out byte[] buffer)
 
             semaphore.Wait(WaitForReadDataTimeoutMilliseconds);
 
-            bool dequeueSucceeded = packetQueue.TryDequeue(out buffer);
+            bool dequeueSucceeded = false;
+            int remainingRetries = 3;
```

Comment:
Nit: since this appears to be the total number of tries rather than retries, maybe ```remainingTries```